### PR TITLE
add rootstock hardhat starter kit

### PR DESCRIPTION
**Pull request by wise4rmgodroot**

### CODE REVIEW

File: docs/04-resources/04-tutorials/rootstock-hardhat-tarter-kit.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
+---
+sidebar_lable: Rootstock Hardhat Starter Kit!
```
Comment (owans) [inline docs/04-resources/04-tutorials/rootstock-hardhat-tarter-kit.md]:
```suggestion
sidebar_label: Quick Start with Hardhat
```

File: docs/04-resources/04-tutorials/rootstock-hardhat-tarter-kit.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+sidebar_position: 6
+title: Rootstock Hardhat Starter Kit! 
+description: 'Whether you are a seasoned developer or just starting your journey into smart contract development, this kit provides a solid foundation for building decentralized applications (DApps) on the RSK network.'
```
Comment (owans) [inline docs/04-resources/04-tutorials/rootstock-hardhat-tarter-kit.md]:
```suggestion
description: 'Whether you are a seasoned developer or just starting your journey into smart contract development, the hardhat starter kit provides a solid foundation for building decentralized applications (dApps) on the Rootstock network.'
```

File: docs/04-resources/04-tutorials/rootstock-hardhat-tarter-kit.md [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+tags: [rsk, rootstock, tutorials, resources, wagmi, starter kit, etherspot, react, dApp, abstraction]
```
Comment (owans) [inline docs/04-resources/04-tutorials/rootstock-hardhat-tarter-kit.md]:
```suggestion
tags: [rsk, rootstock, tutorials, resources, hardhat, starter kit, etherspot, dApp, smart contracts]
```

File: docs/04-resources/04-tutorials/rootstock-hardhat-tarter-kit.md [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+---
+
+
+Whether you’re a seasoned developer or just starting your journey into smart contract development, this kit provides a solid foundation for building decentralized applications (DApps) on the RSK network.
```
Comment (owans) [inline docs/04-resources/04-tutorials/rootstock-hardhat-tarter-kit.md]:
```suggestion
Whether you’re a seasoned developer or just starting your journey into smart contract development, the hardhat starter kit provides a solid foundation for building decentralized applications (dApps) on the Rootstock network.
```

File: docs/04-resources/04-tutorials/rootstock-hardhat-tarter-kit.md [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+
+## What Is RSK?
```
Comment (owans) [inline docs/04-resources/04-tutorials/rootstock-hardhat-tarter-kit.md]:
```suggestion
## What Is Rootstock?
```

File: docs/04-resources/04-tutorials/rootstock-hardhat-tarter-kit.md [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+Rootstock (RSK) is fully compatible with Ethereum. It brings the power of smart contracts to Bitcoin, allowing developers to leverage Bitcoin’s security while benefiting from Ethereum’s ecosystem.
```
Comment (owans) [inline docs/04-resources/04-tutorials/rootstock-hardhat-tarter-kit.md:13]:
```suggestion
Rootstock is fully evm-compatible Bitcoin L2. It brings the power of smart contracts to Bitcoin, allowing developers to leverage Bitcoin’s security while benefitting from Ethereum’s smart contract capabilities.
```

File: docs/04-resources/04-tutorials/rootstock-hardhat-tarter-kit.md [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+
+## Rootstock Hardhat Starter Kit Configuration
```
Comment (owans) [inline docs/04-resources/04-tutorials/rootstock-hardhat-tarter-kit.md:15]:
```suggestion
## Hardhat Configuration
```

File: docs/04-resources/04-tutorials/rootstock-hardhat-tarter-kit.md [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+5. Best Practices Enforcement:
+    - [Solhint](https://github.com/protofire/solhint) is configured to enforce best practices in your Solidity code.
+    - Helps maintain code quality and adherence to recommended patterns.
+
+6. Prepared Examples and Tasks:
+    - Includes examples, tests, deployments, and tasks for common ERC standards (ERC20, ERC721, ERC1155).
+    - Jumpstart your development by leveraging these templates.
+
+## Supported Networks
+We support the following networks for this starter kit:
```
Comment (owans) [inline docs/04-resources/04-tutorials/rootstock-hardhat-tarter-kit.md:40]:
```suggestion
The following networks are supported:
```

File: docs/04-resources/04-tutorials/rootstock-hardhat-tarter-kit.md [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+    - Prettier ensures consistent code formatting, while ESLint catches potential issues.
+
+5. Best Practices Enforcement:
+    - [Solhint](https://github.com/protofire/solhint) is configured to enforce best practices in your Solidity code.
+    - Helps maintain code quality and adherence to recommended patterns.
+
+6. Prepared Examples and Tasks:
+    - Includes examples, tests, deployments, and tasks for common ERC standards (ERC20, ERC721, ERC1155).
+    - Jumpstart your development by leveraging these templates.
+
```
Comment (owans) [inline docs/04-resources/04-tutorials/rootstock-hardhat-tarter-kit.md:38]: Please take out the above content, leave on the README. Move supported networks below prerequisites

File: docs/04-resources/04-tutorials/rootstock-hardhat-tarter-kit.md [hunk 10]
```diff
@@ -0,0 +1,166 @@
...
+We support the following networks for this starter kit:
+1. **Hardhat Network (localhost):** 
+    - Test your smart contracts and dApp functionality on the Hardhat local network.
+2. **Rootstock Mainnet:** 
+    - Deploy your dApp on the Rootstock Mainnet for production use.
+3. **Rootstock Testnet:** 
+    - Test your smart contracts and dApp functionality on the Rootstock testnet before deploying to the mainnet.
+
+## Prerequisites
+Before starting the Rootstock Hardhat starter kit, make sure you have the following prerequisites in place:
```
Comment (owans) [inline docs/04-resources/04-tutorials/rootstock-hardhat-tarter-kit.md:49]:
```suggestion
Before starting the Rootstock Hardhat starter kit, make sure you have the following prerequisites installed:
```

File: docs/04-resources/04-tutorials/rootstock-hardhat-tarter-kit.md [hunk 11]
```diff
@@ -0,0 +1,166 @@
...
+cd rootstock-hardhat-starterkit
+```
+Install all required dependencies using npm:
+
+```bash
+npm install
+```
+
+### Obtain Rootstock Testnet and Mainnet RPC URLs
+This section will walk you through adding Rootstock (RSK) Testnet and Mainnet RPC URLs to your development environment. These URLs are essential for connecting your application to the RSK network and interacting with smart contracts.
```
Comment (owans) [inline docs/04-resources/04-tutorials/rootstock-hardhat-tarter-kit.md:99]:
```suggestion
This section will walk you through adding Rootstock Testnet and Mainnet RPC URLs to your development environment. These URLs are essential for connecting your application to the Rootstock network and interacting with smart contracts.
```

File: docs/04-resources/04-tutorials/rootstock-hardhat-tarter-kit.md [hunk 12]
```diff
@@ -0,0 +1,166 @@
...
+
+There are two ways to obtain RPC URLs:
+
+#### Using Public RPC URLs:
+- Visit the [MetaMask Integration on the Rootstock Dev Portal](https://developers.rsk.co/wallet/use/metamask/). This guide provides instructions on setting up MetaMask for RSK. While following these steps, pay close attention to the sections on adding custom networks. You'll find the RPC URLs for RSK Testnet and Mainnet listed there.
```
Comment (owans) [inline docs/04-resources/04-tutorials/rootstock-hardhat-tarter-kit.md]:
```suggestion
- Visit the [MetaMask Integration on the Rootstock Dev Portal](https://developers.rsk.co/wallet/use/metamask/). This guide provides instructions on setting up MetaMask for Rootstock. While following these steps, pay close attention to the sections on adding custom networks. You'll find the RPC URLs for Rootstock Testnet and Mainnet listed.
```

File: docs/04-resources/04-tutorials/rootstock-hardhat-tarter-kit.md [hunk 13]
```diff
@@ -0,0 +1,166 @@
...
+
+#### Using Hosted RPC URLs (GetBlock):
+- Create an account at the [GetBlock website](https://getblock.io/). Once logged in, navigate to your dashboard.
```
Comment (owans) [inline docs/04-resources/04-tutorials/rootstock-hardhat-tarter-kit.md:107]: Why is RPC API not used here?

Comment (owans) [reply docs/04-resources/04-tutorials/rootstock-hardhat-tarter-kit.md:107]: Replace GetBlock with RPC API and use the Mainnet and Testnet RPC endpoints

File: docs/04-resources/04-tutorials/rootstock-hardhat-tarter-kit.md [hunk 14]
```diff
@@ -0,0 +1,166 @@
...
+The above command will return an output similar to the following:
+```
+Generating typings for: 36 artifacts in dir: typechain-types for target: ethers-v6
+Successfully generated 106 typings!
+Compiled 34 Solidity files successfully (evm target: paris).
+deploying "MockERC721" (tx: 0x9ad1dbc047b78594cf2cad105ded54c851fc0895ae69e4381908fecedd0ee3fc)...: deployed at 0x2E027a3a05f3de6777B23397a50a60ecd04fe34C with 2849621 gas
+```
+
+## Interacting with the Contract - Minting a Token
+Once your contract is deployed, you can interact with it using Hardhat's `erc721-mint` command. This command allows you to mint (create) new ERC721 tokens.
```
Comment (owans) [inline docs/04-resources/04-tutorials/rootstock-hardhat-tarter-kit.md]:
```suggestion
On contract deployment, you can interact with it using Hardhat's `erc721-mint` command. This command allows you to mint (create) new ERC721 tokens.
```

File: docs/04-resources/04-tutorials/rootstock-hardhat-tarter-kit.md [hunk 15]
```diff
@@ -0,0 +1,166 @@
...
+
+### Minting a Token:
+In your terminal, run the following command, replacing the placeholders with your actual values:
```
Comment (owans) [inline docs/04-resources/04-tutorials/rootstock-hardhat-tarter-kit.md]:
```suggestion
In your terminal, run the following command, replacing the placeholders with actual values:
```

File: docs/04-resources/04-tutorials/rootstock-hardhat-tarter-kit.md [hunk 16]
```diff
@@ -0,0 +1,166 @@
...
+```
+Example command:
+```bash
+hh erc721-mint --contract 0x2E027a3a05f3de6777B23397a50a60ecd04fe34C --recipient 0xB0f22816750851D18aD9bd54c32C5e09D1940F7d --network rskTestnet
+```
+- `<ContractAddress>`: Replace this with the address of your deployed ERC721 contract obtained from the previous step.
+- `<RecipientAddress>`: Replace this with the wallet address to receive the newly minted token.
+- `<network>`: Replace this with either `rskTestnet` or `rskMainnet`, depending on the network where your contract is deployed.
+
+This command will initiate a transaction to mint a new ERC721 token and send it to the specified recipient address.
```
Comment (owans) [inline docs/04-resources/04-tutorials/rootstock-hardhat-tarter-kit.md]:
```suggestion
This command will initiate a transaction to mint a new ERC721 tokens and send it to the specified recipient address. 
```

**Comment by owans** [comment]
Please move the guide into the /developers/quickstart/hardhat. @wise4rmgodroot

File: docs/04-resources/04-tutorials/rootstock-hardhat-tarter-kit.md (hunk 13 above)
Comment (wise4rmgodroot) [reply docs/04-resources/04-tutorials/rootstock-hardhat-tarter-kit.md:107]: I think we neeed to confirm from the engineers if this https://dashboard.rpc.rootstock.io/dashboard can replace GetBlock and also our RPC API comes with a private key and might alter most part of the guide, maybe we can revisit this later after launching.
I think you can merge it for now, then we discuss  removing it.

**Review by owans — APPROVED** [review]

**Pull request merged**